File: javascript/rotational-cipher/rotational-cipher.js
// @ts-check

const ord = (ch='') => ch.charCodeAt(0), chr = String.fromCharCode,
      isUpper = (ch='') => !(ord(ch) & 0x20), LEN = 26, A = ord('A'),
      ABC = Array.from({ length: LEN }, (_, i) => chr(i + A)).join('');

export function rotate(chars='', rot=0, cypher='') {
  for (const ch of chars) {
    const idx = ABC.indexOf(ch.toUpperCase()), char = ABC[(idx + rot) % LEN];
    cypher += ~idx ? isUpper(ch) && char || char.toLowerCase() : ch; }
  return cypher; }

File: python/rotational-cipher/rotational_cipher.py
rotate = lambda t='', r=0: ''.join(c.isalpha() and _abc(c, r) or c for c in t)

def _abc(c: str, rot: int, UP=ord('A'), LO=ord('a')):
    return chr((ord(c.upper()) - UP + rot) % 26 + (c.isupper() and UP or LO))
